### PR TITLE
[daint] GDAL-3.4.1-CrayGNU-21.09 with depenencies

### PR DESCRIPTION
**Pull request by gkuhl**
Add GDAL and dependencies (expat, GEOS, HDF and libgeotiff) for CrayGNU-21.09.

### CODE REVIEW

**Comment by lucamar** [comment]
Ok to test

File: easybuild/easyconfigs/h/HDF/HDF-4.2.15-CrayGNU-21.09.eb [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+toolchain = {'name': 'CrayGNU', 'version': '21.09'}
+toolchainopts = {'pic': True}
+
+# https://support.hdfgroup.org/ftp/HDF/releases/HDF4.2.15/src/hdf-4.2.15.tar.gz
+source_urls = ['https://support.hdfgroup.org/ftp/%(name)s/releases/HDF4.2.15/src']
+sources = [SOURCELOWER_TAR_GZ]
+checksums = ['dbeeef525af7c2d01539906c28953f0fdab7dba603d1bc1ec4a5af60d002c459']
+
+builddependencies = [
+    ('Bison', '3.3.2'),
```
Comment (kraushm) [inline easybuild/easyconfigs/h/HDF/HDF-4.2.15-CrayGNU-21.09.eb:22]:
```suggestion
    ('Bison', '3.8.2'),
```

File: easybuild/easyconfigs/h/HDF/HDF-4.2.15-CrayGNU-21.09.eb [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+description = """
+ HDF (also known as HDF4) is a library and multi-object file format for
+ storing and managing data between machines.
+"""
+
+toolchain = {'name': 'CrayGNU', 'version': '21.09'}
+toolchainopts = {'pic': True}
+
+# https://support.hdfgroup.org/ftp/HDF/releases/HDF4.2.15/src/hdf-4.2.15.tar.gz
+source_urls = ['https://support.hdfgroup.org/ftp/%(name)s/releases/HDF4.2.15/src']
```
Comment (kraushm) [inline easybuild/easyconfigs/h/HDF/HDF-4.2.15-CrayGNU-21.09.eb:17]:
```suggestion
source_urls = ['https://support.hdfgroup.org/ftp/%(name)s/releases/%(name)s%(version)s/src']
```

**Review by kraushm — CHANGES_REQUESTED** [review]
Thanks for the PR @gkuhl.

The suggested changes are only to remove hard coded versions and to reuse already built dependencies.

**Review by kraushm — APPROVED** [review]
lgtm. Thanks!

**Pull request merged**